lib/GenomeFileUtil/GenomeFileUtilImpl.py

```diff
@@ -32,6 +32,31 @@ def __init__(self, config):
         self.re_api_url = config['re-api-url']
         self.raw = config
 
+
+# Helper function to validate max_threads catalog param
+def _validate_max_threads_type(threads_count, var_name, default_val):
```

Comment:
The functions _validate_max_threads_type and _validate_threads_per_cpu_type are nearly identical, differing only in the variable type they cast. Combining them and introducing an 'expected_type' argument might streamline their functionality.

Reply:
Updated!